xiplot/tabs/plugins.py

```diff
@@ -0,0 +1,477 @@
+            disabled=True,
+        )
+
+        return FlexRow(
+            html.Div(
+                [
+                    layout_wrapper(
+                        component=FlexRow(
+                            dcc.Dropdown(
+                                get_suggested_plugin_options(
+                                    plugin_dir=plugin_dir
+                                ),
+                                id="plugin_paths",
+                                placeholder=(
+                                    "Select a suggested plugin or type its URL"
+                                    " or PyPi package name"
+                                ),
+                            ),
+                            layout_wrapper(
+                                component=dcc.Input(id="plugin-search-input"),
+                                style={"display": "none"},
+                            ),
+                            html.Button(
+                                "Install",
+                                id="plugin-submit-button",
+                                n_clicks=0,
+                                className="button",
+                            ),
+                        ),
+                        title="Install a Plugin",
+                        css_class="dash-dropdown",
+                    ),
+                    html.Br(),
+                    layout_wrapper(
+                        component=FlexRow(
+                            dcc.Dropdown(
+                                get_loaded_plugin_options(),
+                                id="loaded_plugins",
+                                clearable=False,
+                                searchable=False,
+                                placeholder=(
+                                    "Check the list of already loaded plugins"
+                                ),
+                            ),
+                            html.Button(
+                                "Reload χiplot",
+                                id="plugin-reload-button",
+                                n_clicks=0,
+                                className="button",
+                                disabled=True,
+                            ),
+                            dmc.Modal(
+                                title="Do you really want to reload χiplot?",
+                                id="plugin-reload-modal",
+                                zIndex=10000,
+                                children=[
+                                    dmc.Text(
+                                        "Please ensure that you have"
+                                        " downloaded your plots and data so"
+                                        " that nothing is lost. Any plugins"
+                                        " that you have installed will become"
+                                        " loaded after the reload."
+                                    ),
```

Comment:
A list of the packages that would become activated would be nice (here or somewhere else). But that could also be left for future work.

Reply:
Good idea - I'll work on this tomorrow and then merge the PR